ext/nnstreamer/tensor_decoder/tensordec-boundingbox.h

```diff
@@ -0,0 +1,607 @@
+        y_center = (box[0] * 1.f) / i_height * a->h + a->y_center;                      \
+        x_center = (box[1] * 1.f) / i_width * a->w + a->x_center;                       \
+        h = (box[2] * 1.f) / i_height * a->h;                                           \
+        w = (box[3] * 1.f) / i_width * a->w;                                            \
+        ymin = y_center - h / 2.f;                                                      \
+        xmin = x_center - w / 2.f;                                                      \
+        y = ymin * i_height;                                                            \
+        x = xmin * i_width;                                                             \
+        width = w * i_width;                                                            \
+        height = h * i_height;                                                          \
+        object.class_id = 0;                                                            \
+        object.x = MAX (0, x);                                                          \
+        object.y = MAX (0, y);                                                          \
+        object.width = width;                                                           \
+        object.height = height;                                                         \
+        object.prob = score;                                                            \
+        object.valid = TRUE;                                                            \
+        g_array_append_val (results, object);                                           \
+      }                                                                                 \
+    }                                                                                   \
+    break
+
+/** @brief Macro to simplify calling _get_objects_mp_palm_detection */
+#define _get_objects_mp_palm_detection_(type, typename) \
+  _get_objects_mp_palm_detection (type, typename, (detections->data), (boxes->data), config)
+
+class BoxProperties
+{
+  public:
+  virtual ~BoxProperties () = default;
+
+  /* mandatory methods */
+  virtual int setOptionInternal (const char *param) = 0;
+  virtual int checkCompatible (const GstTensorsConfig *config) = 0;
+  virtual GArray *decode (const GstTensorsConfig *config, const GstTensorMemory *input) = 0;
+
+  void setInputWidth (guint width)
+  {
+    i_width = width;
+  }
+  void setInputHeight (guint height)
+  {
+    i_height = height;
+  }
+  void setTotalLabels (guint labels)
+  {
+    total_labels = labels;
+  }
+
+  guint getInputWidth ()
+  {
+    return i_width;
+  }
+  guint getInputHeight ()
+  {
+    return i_height;
+  }
+
+  protected:
+  guint i_width; /**< Input Video Width */
+  guint i_height; /**< Input Video Height */
+
+  guint max_detection;
+  guint total_labels;
+};
+
+
+class BoundingBox
+{
+  public:
+  BoundingBox ();
+  ~BoundingBox ();
```

Comment:
You will need virtual destructors unless you can guarantee that subclasses don't have dynamically allocated properties.